client/my-sites/plugins/plugins-list/index.jsx

```diff
@@ -0,0 +1,390 @@
+/**
+ * External dependencies
+ */
+import React, { PropTypes } from 'react';
+import classNames from 'classnames';
+import isEmpty from 'lodash/lang/isEmpty';
+
+/**
+ * Internal dependencies
+ */
+import analytics from 'analytics';
+import acceptDialog from 'lib/accept';
+import DisconnectJetpackDialog from 'my-sites/plugins/disconnect-jetpack/disconnect-jetpack-dialog';
+import PluginItem from 'my-sites/plugins/plugin-item/plugin-item';
+import PluginsActions from 'lib/plugins/actions';
+import PluginsListHeader from 'my-sites/plugins/plugin-list-header';
+import PluginsLog from 'lib/plugins/log-store';
+import PluginNotices from 'lib/plugins/notices';
+import SectionHeader from 'components/section-header';
+
+export default React.createClass( {
+	displayName: 'PluginsList',
+
+	mixins: [ PluginNotices ],
+
+	propTypes: {
+		plugins: PropTypes.array,
+		header: PropTypes.string,
+		isWpCom: PropTypes.bool,
+		sites: PropTypes.object,
+		selectedSite: PropTypes.oneOfType( [ PropTypes.bool, PropTypes.object ] ),
+		pluginUpdateCount: PropTypes.number,
+		isPlaceholder: PropTypes.bool,
+	},
+
+	getInitialState() {
+		return {
+			disconnectJetpackDialog: false,
+			bulkManagement: false,
+			selectedPlugins: {}
+		};
+	},
+
+	componentDidMount() {
+		PluginsLog.on( 'change', this.showDisconnectDialog );
+	},
+
+	componentWillUnmount() {
+		PluginsLog.removeListener( 'change', this.showDisconnectDialog );
+	},
+
+	isSelected( plugin ) {
+		return !! this.state.selectedPlugins[ plugin.slug ];
+	},
+
+	togglePlugin( plugin ) {
+		let selectedPlugins = Object.assign( {}, this.state.selectedPlugins );
+		selectedPlugins[ plugin.slug ] = ! this.state.selectedPlugins[ plugin.slug ];
+		this.setState( { selectedPlugins } );
+		analytics.ga.recordEvent( 'Plugins', 'Clicked to ' + this.isSelected( plugin ) ? 'Deselect' : 'Select' + 'Single Plugin', 'Plugin Name', plugin.slug );
+	},
+
+	setBulkSelectionState( plugins, selectionState ) {
+		let slugsToBeUpdated = {};
+		plugins.forEach( plugin => slugsToBeUpdated[ plugin.slug] = selectionState );
+
+		let selectedPlugins = Object.assign( {}, this.state.selectedPlugins, slugsToBeUpdated );
+
+		this.setState( { selectedPlugins } );
+	},
+
+	getPluginBySlug( slug ) {
+		for ( let i = 0, l = this.props.plugins.lenght; i < l; i++ ) {
+			if ( this.props.plugins[ i ].slug === slug ) {
+				return this.props.plugins[ i ];
+			}
+		}
+		return false;
+	},
```

Comment:
`Array.prototype.find()` I am not sure if it is something that is polyfilled by bubble? Since it doesn't seem to be supported yet accross all browsers (https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/find ) 

Is there a way to check that. Maybe using the `lodash` equivalent would be best?